bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/util/BundleResolver.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.util;
+
+import org.osgi.framework.Bundle;
+
+/**
+ * Resolve bundle specific information from the framework.
+ *
+ * @author Henning Treu - initial contribution
+ *
+ */
+public interface BundleResolver {
+
+    /**
+     * Resolve the bundle name associated with the given {@link Class}.
+     *
+     * @param clazz the {@link Class} to resolve the bundle name for.
+     * @return the bundle name associated with the given {@link Class}.
+     */
+    String resolveSymbolicName(Class<?> clazz);
```

Comment:
Do we really need that function in the `BundleResolver`?
Why not using the `resolveBundle` in the interface only and the one that is interested in the symbolic name calls `getSymbolicName()` on the returned bundle itself?

The test mocks could return a mock bundle that `getSymbolicName()` that return `BUNDLE_NAME` (WRT below).

If we provide that separate functions one user of that interface could call `resolveBundle(someClass).getSymbolicName()`, the other one calls `resolveSymbicName()`.
As long as the methods return the same string, it does not matter, but the question would be why adding this redundant method then.
If the methods return different strings sometime, then we have also to provide a function `resolveSymbolicName(Bundle bundle)` that must be used instead of `bundle.getSymbolicName()`...

Reply:
Yes, agreed. I will change the PR accordingly.